File: nice-traffic-light/r/R/presenter.R
# Write a traffic_light_report to a text sink.

#' Write a traffic_light_report to a writable connection.
#'
#' @param report A `traffic_light_report`.
#' @param sink A writable connection (e.g. `stdout()`).
present <- function(report, sink) {
  cat(sprintf("%-6s %s\n", report$colour, report$headline), file = sink)
  for (trigger in report$outcome$triggers) {
    cat(sprintf("  - %s\n", trigger), file = sink)
  }
  invisible(NULL)
}

File: nice-traffic-light/node/src/presenter.ts
import type { TrafficLightReport } from './trafficLight.js';

export interface TextSink {
  write(text: string): void;
}

export function present(report: TrafficLightReport, sink: TextSink): void {
  sink.write(`${report.colour.padEnd(6, ' ')} ${report.headline}\n`);
  for (const trigger of report.outcome.triggers) {
    sink.write(`  - ${trigger}\n`);
  }
}
